backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/log/ModularContractLogDAO.java

```diff
@@ -240,4 +253,22 @@ private static Quantity extractQty(@NonNull final I_ModCntr_Log record)
 	{
 		return Quantitys.create(record.getQty(), UomId.ofRepoId(record.getC_UOM_ID()));
 	}
+
+	public Optional<ModularContractLogEntry> getLastModularContractLog(
+			@NonNull final FlatrateTermId modularFlatrateTermId,
+			@NonNull final OrderLineId orderLineId)
+	{
+		final TableRecordReference modularRecordReference = TableRecordReference.of(I_C_OrderLine.Table_Name, orderLineId);
```

Comment:
why not using lastRecord(query)?

Reply:
didn't exist like that when i implemented it